input/system/rds/logs.go

```diff
@@ -182,71 +152,9 @@ func downloadRdsLogFilePortion(rdsSvc *rds.RDS, tmpFile *os.File, logger *util.L
 		return
 	}
 
-	if len(*resp.LogFileData) > 0 {
-		newBytesWritten, err = tmpFile.WriteString(*resp.LogFileData)
-		if err != nil {
-			err = fmt.Errorf("Error writing to tempfile: %s", err)
-			return
-		}
-	}
-
+	content = resp.LogFileData
 	newMarker = resp.Marker
 	additionalDataPending = *resp.AdditionalDataPending
 
 	return
 }
-
-// Analyze and submit at most the trailing 10 megabytes of the retrieved RDS log file portions
-//
-// This avoids an OOM in two edge cases:
-// 1) When starting the collector, as we always load the last 10,000 lines (which may be very long)
-// 2) When extremely large values are output in a single log event (e.g. query parameters in a DETAIL line)
-//
-// We intentionally throw away data here (and warn the user about it), since the alternative
-// is often a collector crash (due to OOM), which would be less desirable.
-const maxLogParsingSize = 10 * 1024 * 1024
```

Comment:
I suspect this is no longer needed as of #384 and #597, though this will require production testing to confirm.

Reply:
Hmm, it seems the 10k requests on boot is unavoidable because of [API limitations](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_DownloadDBLogFilePortion.html). I wonder if instead of using an arbitrary cutoff, we can ask the operating system how much free memory there is, and only use half of that?